src/movies.js

```diff
@@ -1,28 +1,110 @@
+  return moviesArray.map(movie => movie.director);
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  const moviesFiltered = moviesArray.filter(movie => (movie.director == 'Steven Spielberg' && movie.genre.includes("Drama")))
+  return moviesFiltered.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  if (moviesArray.length>0) {
+    const moviesWithScores = moviesArray.filter(movie => typeof movie.score == 'number'); 
+    const sumScores = moviesWithScores.reduce(function (accumulator, currentValue) {
+      return accumulator + currentValue.score;
+    }, 0);
+    return Math.round(sumScores/moviesArray.length*100)/100;
+  }
+  else {
+    return 0;
+  }
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+  const moviesFiltered = moviesArray.filter(movie => (movie.genre.includes("Drama") && typeof movie.score == 'number'));
+    const sumScores = moviesFiltered.reduce(function (accumulator, currentValue) {
+    return accumulator + currentValue.score;
+  }, 0);
+  if (moviesFiltered.length > 0) {
+    return Math.round(sumScores/moviesFiltered.length*100)/100;
+  }
+  else {
+    return 0;
+  }
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+  const moviesArrayCopy = [...moviesArray];
+  const sortedByYear = moviesArrayCopy.sort(function(movie1, movie2) {
+    return movie1.year - movie2.year;
+  })
+  const sortedByYearAndAlph = sortedByYear.sort(function(movie1, movie2) {
+    if (movie1.year == movie2.year) {
+      return movie1.title.localeCompare(movie2.title);      
+    }
+  })
+  return sortedByYearAndAlph;
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
+function orderAlphabetically(moviesArray) {
+  const moviesArrayCopy = [...moviesArray];
+  const titles = moviesArrayCopy.map(movie => movie.title);
+  const titlesSortedByAlph = titles.sort();
+  return titlesSortedByAlph.splice(0,20);
+
+}
 
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
-function turnHoursToMinutes(moviesArray) {}
+function turnHoursToMinutes(moviesArray) {
+  const moviesArrayCopy = [...moviesArray];
+
+  moviesArrayCopy.forEach(function (element) {
+    let hours = 
+    element.duration =
+      parseInt(element.duration.slice(0, 1)) * 60 +
+      parseInt(element.duration.slice(3, 5));
+  });
+  return moviesArrayCopy;
+}
 
 // BONUS - Iteration 8: Best yearly score average - Best yearly score average
-function bestYearAvg(moviesArray) {}
+
```

Comment:
I think that in this iteration would be better try to maintain the relationship of years and scores in the same element